src/main/java/org/opentripplanner/model/Timetable.java

```diff
@@ -133,49 +141,56 @@ public TripTimes setTripTimes(int tripIndex, TripTimes tt) {
    *
-   * @param tripUpdate        GTFS-RT trip update
-   * @param timeZone          time zone of trip update
-   * @param updateServiceDate service date of trip update
+   * @param tripUpdate                    GTFS-RT trip update
+   * @param timeZone                      time zone of trip update
+   * @param updateServiceDate             service date of trip update
    * @param backwardsDelayPropagationType Defines when delays are propagated to previous stops and
-   *                          if these stops are given the NO_DATA flag
-   * @return {@link TripTimesPatch} that contains a new copy of updated TripTimes after TripUpdate
-   * has been applied on TripTimes of trip with the id specified in the trip descriptor of the
-   * TripUpdate and a list of stop indices that have been skipped with the realtime update; null if
-   * something went wrong
+   *                                      if these stops are given the NO_DATA flag
+   * @return {@link Result<TripTimesPatch, UpdateError>} contains either a new copy of updated
+   * TripTimes after TripUpdate has been applied on TripTimes of trip with the id specified in the
+   * trip descriptor of the TripUpdate and a list of stop indices that have been skipped with the
+   * realtime update; or an error if something went wrong
    * <p>
    * TODO OTP2 - This method depend on GTFS RealTime classes. Refactor this so GTFS RT can do
    *           - its job without sending in GTFS specific classes. A generic update would support
    *           - other RealTime updats, not just from GTFS.
    */
-  public TripTimesPatch createUpdatedTripTimes(
+  public Result<TripTimesPatch, UpdateError> createUpdatedTripTimes(
     TripUpdate tripUpdate,
     ZoneId timeZone,
     LocalDate updateServiceDate,
     BackwardsDelayPropagationType backwardsDelayPropagationType
   ) {
+    Result<TripTimesPatch, UpdateError> generalError = Result.failure(
+      UpdateError.noTripId(UNKNOWN)
+    );
     if (tripUpdate == null) {
-      LOG.error("A null TripUpdate pointer was passed to the Timetable class update method.");
-      return null;
+      LOG.debug("A null TripUpdate pointer was passed to the Timetable class update method.");
+      return generalError;
     }
 
     // Though all timetables have the same trip ordering, some may have extra trips due to
     // the dynamic addition of unscheduled trips.
     // However, we want to apply trip updates on top of *scheduled* times
     if (!tripUpdate.hasTrip()) {
-      LOG.error("TripUpdate object has no TripDescriptor field.");
-      return null;
+      LOG.debug("TripUpdate object has no TripDescriptor field.");
+      return generalError;
     }
 
     TripDescriptor tripDescriptor = tripUpdate.getTrip();
     if (!tripDescriptor.hasTripId()) {
-      LOG.error("TripDescriptor object has no TripId field");
-      return null;
+      LOG.debug("TripDescriptor object has no TripId field");
+      Result.failure(UpdateError.noTripId(UNKNOWN));
```

Comment:
`TRIP_ID_NOT_FOUND` Or a more specific error about the fuzzy trip matching failing